haystack/nodes/prompt/prompt_node.py

```diff
@@ -840,19 +849,48 @@ def is_supported_template(self, prompt_template: Union[str, PromptTemplate]) ->
         template_name = prompt_template if isinstance(prompt_template, str) else prompt_template.name
         return template_name in self.prompt_templates
 
-    def get_prompt_template(self, prompt_template: Union[str, PromptTemplate, None]) -> PromptTemplate:
+    def get_prompt_template(self, prompt_template: Union[str, PromptTemplate, None] = None) -> Optional[PromptTemplate]:
         """
-        Returns a prompt template by name.
-        :param prompt_template_name: The name of the prompt template to be returned.
-        :return: The prompt template object.
+        Resolves a prompt template.
+
+        :param prompt_template: The prompt template to be resolved. You can choose between the following types:
+            - None: returns the default prompt template
+            - PromptTemplate: returns the given prompt template object
+            - str: parses the string depending on its content:
+                - prompt template name: returns the prompt template registered with the given name
+                - prompt template yaml: returns a prompt template specified by the given yaml
+                - prompt text: returns a copy of the default prompt template with the given prompt text
+
+            :return: The prompt template object.
         """
+        prompt_template = prompt_template or self.default_prompt_template
+        if prompt_template is None:
+            return None
```

Comment:
Returning `None` instead of throwing an exception makes it easier to use. That doesn't mean we're not raising any exceptions any more. We raise if we're sure there's something wrong. Look below.